File: GolangAlgorithm/Week03/Recursion/104/maxDepth_test.go
package maxDepth

import (
	"Week03/Tree/BinaryTree"
	"fmt"
	"testing"
)

func TestMaxDepth(t *testing.T) {
	root := &BinaryTree.TreeNode{Val: 5}
	root.Left = &BinaryTree.TreeNode{Val: 4}
	root.Right = &BinaryTree.TreeNode{Val: 6}
	root.Right.Left = &BinaryTree.TreeNode{Val: 3}
	root.Right.Right = &BinaryTree.TreeNode{Val: 7}
	fmt.Println(dfs(root))
}

File: JavascriptAlgorithm/Test/Tree/maxDepthTest.js
import { maxDepth } from "../../Tree/maxDepth-104.js";
import assert from "assert";
import { TreeNode } from "../../Tree/TreeNode.js";
describe("maxDepth", function () {
  function makeTest(tree, res) {
    it(`[${JSON.stringify(tree)}]'s maxDepth is ${JSON.stringify(
      res
    )}`, function () {
      assert.equal(JSON.stringify(maxDepth(tree)), JSON.stringify(res));
    });
  }
  let tree1 = new TreeNode(1);
  tree1.left = new TreeNode(2);
  tree1.right = new TreeNode(3);

  let tree2 = new TreeNode(2);
  tree2.left = new TreeNode(1);
  tree2.right = new TreeNode(3);

  let tree3 = new TreeNode(5);
  tree3.left = new TreeNode(1);
  tree3.right = new TreeNode(4);
  tree3.right.left = new TreeNode(3);
  tree3.right.right = new TreeNode(6);

  let res = [2, 2, 3];
  let trees = [tree1, tree2, tree3];
  for (let i = 0; i < trees.length; i++) {
    makeTest(trees[i], res[i]);
  }
});
